### ISSUE TITLE
[v3] Implement new Concept Exercise: constructors

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#845]
This issue describes how to implement the `constructors` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Constructors in F#.

## Learning objectives

- Know what constructors are.
- Know the difference between primary and additional constructors.
- Know how to define additional constructors.
- Know how to add functionality to constructors using  `do` bindings.
- Know that constructors are regular functions.

## Out of scope

- Static `do` bindings.

## Concepts

The Concepts this exercise unlocks are:

- `constructors`: know what constructors are; know the difference between primary and additional constructors; know how to define additional constructors; know how to add functionality to constructors using  `do` bindings; know that constructors are regular functions.

## Prerequisites

This exercise's prerequisites Concepts are:

- `classes`: know what records are.
- `method-overloading`: know how method overloading works.

Any data types used in this exercise (e.g. `strings` or `integers`) should also be added as prerequisites.

### Hints

- [Constructors][constructors]: introduction to constructors.
- [`do` bindings in constructors][do-bindings-in-classes]: how to use `do` bindings to add constructor logic.
- [Additional constructors][additional-constructors]: how to define additional constructors.

### After

- [Constructors][constructors]: introduction to constructors.
- [`do` bindings in constructors][do-bindings-in-classes]: how to use `do` bindings to add constructor logic.
- [Additional constructors][additional-constructors]: how to define additional constructors.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[constructors]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/constructors
[do-bindings-in-classes]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/do-bindings-in-classes
[additional-constructors]: https://fsharpforfunandprofit.com/posts/classes/#secondary-constructors
